contracts/oracles/WamplOracle.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {IOracle} from "../interfaces/IOracle.sol";
+import {IWAMPL} from "../interfaces/IWAMPL.sol";
+import {IChainlinkAggregator} from "../interfaces/IChainlinkAggregator.sol";
+
+/**
+ * @title WamplOracle Oracle
+ *
+ * @notice Provides a value onchain from a chainlink oracle aggregator
+ */
+contract WamplOracle is IOracle {
+    /// @dev The output price has a 8 decimal point precision.
+    uint256 public constant PRICE_DECIMALS = 8;
+    // The address of the Chainlink Aggregator contract
+    IChainlinkAggregator public immutable amplEthOracle;
+    uint256 public immutable amplEthOracleDecimals;
```

Comment:
I wonder if you could just make `amplEthOracleDecimals` and `ethUsdOracleDecimals` constants instead of immutable variables. I don't think these will ever change.

Reply:
I figured this would be best in case testnet versions of contracts have different decimal counts for whatever reason. I can't remember which ones this applies to off the top of my head but there's at least one asset we use that suffers from this issue.